blockstack/blockstackd.py

```diff
@@ -3549,8 +3601,37 @@ def run_blockstackd():
                 print "Database is NOT CONSISTENT with block {} and consensus hash {}".format(block, args.consensus_hash)
                 sys.exit(1)
 
+        print "Querying namespace IDs..."
+        namespaces_entries = db.get_all_namespace_ids()
+        namespaces_entries.sort()
+        namespaces = []
+        for namespace_str in namespaces_entries:
+            namespace_info = db.get_namespace(namespace_str)
+            namespace = {}
+            namespace['namespace_id'] = namespace_info['namespace_id']
+            namespace['address'] = b58ToC32(str(namespace_info['address']))
+            namespaces.append(namespace)
+
+        print "Querying names..."
+        name_entries = db.get_all_names()
+        name_entries.sort()
+        names = []
+        for name_str in name_entries:
+            name_info = load_name_info(db, name_str, block)
+            name = {}
+            name['name'] = name_info['name']
+            name['address'] = b58ToC32(str(name_info['address']))
+            name['expire_block'] = name_info['expire_block']
+            if 'zonefile' not in name_info or name_info['zonefile'] is None:
+                print 'missing zonefile for {}'.format(name)
+                name['zonefile'] = "__MISSING__"
+            else:
+                name['zonefile'] = name_info['zonefile']
+            names.append(name)
```

Comment:
@lgalabru Same question -- anything else we need to export for on-chain names?
Note that several names are missing zonefiles. Should the zonefile hash be exported?

Reply:
The bns contract currently needs the following name fields:
https://github.com/blockstack/stacks-blockchain/blob/09badf4bb7ed570ff649ff18232650eb0c742521/src/chainstate/stacks/boot/sns.clar#L93-L98

Reply:
@lgalabru all relevant field should now be included in the export for names. Expired and/or revoked names are simply omitted from the output.

Reply:
Thank you @zone117x!